skiko/src/commonMain/cpp/common/include/RunRecordClone.hh

```diff
@@ -0,0 +1,63 @@
+// Must match SkTextBlobPriv.h
+//
+// Extended Textblob runs have more data after the Pos[] array:
+//
+//    -------------------------------------------------------------------------
+//    ... | RunRecord | Glyphs[] | Pos[] | TextSize | Clusters[] | Text[] | ...
+//    -------------------------------------------------------------------------
+class RunRecordClone {
+public:
+    SkFont    fFont;
+    uint32_t  fCount;
+    SkPoint   fOffset;
+    uint32_t  fFlags;
+
+    uint16_t* glyphBuffer() const {
+        // Glyphs are stored immediately following the record.
+        return reinterpret_cast<uint16_t*>(const_cast<RunRecordClone*>(this) + 1);
+    }
+
+    SkScalar* posBuffer() const {
+        // Position scalars follow the (aligned) glyph buffer.
+        return reinterpret_cast<SkScalar*>(reinterpret_cast<uint8_t*>(this->glyphBuffer()) +
+                                           SkAlign4(fCount * sizeof(uint16_t)));
+    }
+
+    uint32_t* textSizePtr() const {
+        // textSize follows the position buffer.
+        return (uint32_t*)(&this->posBuffer()[fCount * ScalarsPerGlyph(positioning())]);
+    }
+
+    uint32_t textSize() const {
+        return isExtended() ? *this->textSizePtr() : 0;
+    }
+
+    uint32_t* clusterBuffer() const {
+        // clusters follow the textSize.
+        return isExtended() ? 1 + this->textSizePtr() : nullptr;
+    }
+
+    char* textBuffer() const {
+        return isExtended()
+               ? reinterpret_cast<char*>(this->clusterBuffer() + fCount)
+               : nullptr;
+    }
+
+    uint8_t positioning() const {
+        return fFlags & 0x3; // kPositioning_Mask
+    }
+
+    bool isExtended() const {
+        return fFlags & 0x8; // kExtended_Flag
+    }
+
+    static unsigned ScalarsPerGlyph(uint8_t pos) {
+        const uint8_t gScalarsPerPositioning[] = {
```

Comment:
Why we cannot do that with `switch`?

Reply:
sure. Didn't think about it when copy/pasted from jni.